src/core_plugins/metrics/server/routes/fields.js

```diff
@@ -7,7 +7,12 @@ export default (server) => {
     handler: (req, reply) => {
       return getFields(req)
```

Comment:
hapi route handlers are not promise aware, but could be if you pass the promise to reply instead of returning it. Then the resolve value or error is treated as the argument to reply (like you are doing below).